time_capture/time_capture/doctype/time_capture/time_capture.py

```diff
@@ -214,6 +215,13 @@ def create_timesheets(self):
 				timesheet.insert()
 				timesheet.submit()
 
+	def validate_task_project(self):
+		for log in self.time_logs:
+			if log.task and log.project:
+				task_project = frappe.db.get_value("Task", log.task, "project")
+				if task_project != log.project:
```

Comment:
```suggestion
			if log.project != frappe.db.get_value("Task", log.task, "project"):
```

Little changes: 
- log.task and log.project are mandatory (and probably always will be).
- One line less, if we work without the variable task_project